tensorflow/lite/micro/kernels/if.cc

```diff
@@ -111,32 +197,118 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
+        destination = input->data.raw;
+        source = intermediate_copy_buffer;
+        break;
+        // NO default. enum is owned by this file and new enum shall be handled.
+    }
+    memcpy(destination, source, input_bytes);
+    intermediate_copy_buffer += input_bytes;
   }
+  return kTfLiteOk;
+}
+
+TfLiteStatus CopyFromIfInputToIntermediate(TfLiteContext* context,
+                                           TfLiteNode* node,
+                                           MicroGraph& graph_info,
+                                           int active_branch_subgraph_index) {
+  return CopyInputBetweenIfAndSubgraph(
+      context, node, graph_info, active_branch_subgraph_index,
+      CopyInputDirection::kFromIfToIntermediate);
+}
+
+TfLiteStatus CopyFromIntermediateToIfInput(TfLiteContext* context,
+                                           TfLiteNode* node,
+                                           MicroGraph& graph_info,
+                                           int active_branch_subgraph_index) {
+  return CopyInputBetweenIfAndSubgraph(
+      context, node, graph_info, active_branch_subgraph_index,
+      CopyInputDirection::kFromIntermediateToIf);
+}
+TfLiteStatus CopyFromIntermediateToSubgraphInput(
+    TfLiteContext* context, TfLiteNode* node, MicroGraph& graph_info,
+    int active_branch_subgraph_index) {
+  return CopyInputBetweenIfAndSubgraph(
+      context, node, graph_info, active_branch_subgraph_index,
+      CopyInputDirection::kFromIntermediateToSubgraph);
+}
+
+TfLiteStatus CopyFromIntermediateToIfOutput(TfLiteContext* context,
+                                            TfLiteNode* node,
+                                            MicroGraph& graph_info,
+                                            int active_branch_subgraph_index) {
+  return CopyOutputBetweenIfAndSubgraph(
+      context, node, graph_info, active_branch_subgraph_index,
+      CopyOutputDirection::kFromIntermediateToIf);
+}
+
+TfLiteStatus CopyFromSubgraphOutputToIntermediate(
+    TfLiteContext* context, TfLiteNode* node, MicroGraph& graph_info,
+    int active_branch_subgraph_index) {
+  return CopyOutputBetweenIfAndSubgraph(
+      context, node, graph_info, active_branch_subgraph_index,
+      CopyOutputDirection::kFromSubgraphToIntermediate);
+}
+
+TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
+  const OpData* op_data = reinterpret_cast<OpData*>(node->user_data);
+
+  const TfLiteTensor* cond;
+  TF_LITE_ENSURE_OK(context, GetInputSafe(context, node, 0, &cond));
+  bool cond_value = cond->data.b[0];
+
+  tflite::MicroContext* micro_context = tflite::GetMicroContext(context);
+  MicroGraph& graph_info = micro_context->graph();
+
+  // Currently we copy the input / output between the subgraphs.
+  int active_branch_subgraph_index =
+      cond_value ? op_data->then_subgraph_index : op_data->else_subgraph_index;
+
+  // The input tensors of the IF subgraph and active branch subgraphs are
```

Comment:
Perhaps a clearer comment would be:
"The input tensors of the IF op's subgraph and the active branch subgraph.."